container-investigations/aaa
# Obtain access token through AKS's workload identity mechanism

Test steps:
1. Deploy a AKS pod with a workload identity. See instructions [here](https://learn.microsoft.com/en-us/azure/aks/learn/tutorial-kubernetes-workload-identity) and sample yaml file below.
1. Use `kubectl logs quick-start` to find out the **federated token**
1. Use `kubectl describe pod quick-start` to find out **client id** and **tenant id**
1. invoke `getAccessTokenFromFederatedToken` with 3 env variables: (from step 3) **AZURE_CLIENT_ID**, **AZURE_TENANT_ID**, and (from step 2) **AZURE_FEDERATED_TOKEN_FILE**
```
apiVersion: v1
kind: Pod
metadata:
  name: quick-start
  namespace: default
  labels:
    azure.workload.identity/use: "true"
spec:
  serviceAccountName: workload-identity-sa
  containers:
    - image: alpine
      name: msicontainer
      command: ["cat", "/var/run/secrets/azure/tokens/azure-identity-token"]
  nodeSelector:
    kubernetes.io/os: linux
```
